ncm-spma/src/main/perl/spma/dnf.pm

```diff
@@ -534,19 +534,26 @@ sub Configure
     $self->info("Will skip removal of running kernel: $kvers");
 
     # Compose list of packages to be installed and removed.
-    my $will_remove = $preinstalled - $to_install - "kernel-$kvers" - "kernel-core-$kvers";
+    my $will_remove = $preinstalled - $to_install;
     my $will_install = $to_install - $preinstalled;
     my $whitelist = $t->{whitelist};
     my $whitelisted = Set::Scalar->new();
     for my $rpm ($will_remove->elements) {    # do not remove imported GPG keys
         if (substr($rpm, 0, 11) eq 'gpg-pubkey-') {
             $will_remove->delete($rpm);
         }
+        my $rpm_noepoch = $rpm;
+        $rpm_noepoch =~ s/^.*://;
```

Comment:
My expectation of a variable called `$rpm_noepoch` is that it's the whole thing without an epoch, but looking carefully at the substitution it removes everything before the (last) colon, making the variable more accurately named, `$rpm_version_noepoch` ?

I might be tempted to go for something like this as being a bit more readable than a regex too (but maybe I'm just out of tune with Perl ;-)
```
$rpm_version = (split(':', $rpm))[-1];
```

Reply:
Make sense thanks.